### PR TITLE
CORE-6791: Simulator API KDocs

### PR DESCRIPTION
**Pull request by LizR3**

### CODE REVIEW

**Review by ronanbrowne — DISMISSED** [review]
LGTM -  **(Only build.gradle changes reviewed)**  someone needs to review Kdoc - i presume that's tech writers cc @nadinequinr3

File: simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt [hunk 1]
```diff
@@ -15,12 +23,25 @@ interface HoldingIdentity {
                 throw ServiceConfigurationException(HoldingIdentityFactory::class.java)
         }
 
+        /**
+         * Creates a holding identity using the provided string as a [MemberX500Name] common name, with other
+         * name elements set to defaults.
+         *
+         * @param commonName the string to use as the common name
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt]:
```suggestion
         * @param commonName The string to use as the common name.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt [hunk 2]
```diff
@@ -15,12 +23,25 @@ interface HoldingIdentity {
...
+         * @return a [HoldingIdentity] whose [MemberX500Name] has the given common name
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt]:
```suggestion
         * @return A [HoldingIdentity] whose [MemberX500Name] has the given common name.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt [hunk 3]
```diff
@@ -15,12 +23,25 @@ interface HoldingIdentity {
...
+         */
         fun create(commonName: String): HoldingIdentity {
             return factory.create(
                 MemberX500Name.parse("CN=$commonName, OU=ExampleUnit, O=ExampleOrg, L=London, C=GB")
             )
         }
 
+        /**
+         * Creates a holding identity using the provided member
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt]:
```suggestion
         * Creates a holding identity using the provided member.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt [hunk 4]
```diff
@@ -15,12 +23,25 @@ interface HoldingIdentity {
...
+         *
+         * @param member the member for which to create a holding identity
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt]:
```suggestion
         * @param member The member for which to create a holding identity.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt [hunk 5]
```diff
@@ -15,12 +23,25 @@ interface HoldingIdentity {
...
+         * @return a [HoldingIdentity]
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/HoldingIdentity.kt]:
```suggestion
         * @return A [HoldingIdentity].
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 6]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
 
+/**
+ * A wrapper around [net.corda.v5.application.flows.RPCRequestData] which can be passed to Simulator
+ * on initiation of flows. This interface can be implemented, but static / companion factory methods
+ * are provided for convenience.
+ */
 interface RequestData {
+
+    /**
+     * The client request which would be used by Corda to identify a particular request for a flow call
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
     * The client request which would be used by Corda to identify a particular request for a flow call.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 7]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+     */
     val clientRequestId: String
 
+    /**
+     * The name of the flow class to be run
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
     * The name of the flow class to be run.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 8]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+     */
     val flowClassName: String
 
+    /**
+     * A Json string containing the request body to be passed to the flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
     * A JSON string containing the request body to be passed to the flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 9]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+     */
     val requestBody: String
     fun toRPCRequestData(): RPCRequestData
 
     companion object {
         private val factory = ServiceLoader.load(RequestDataFactory::class.java).firstOrNull() ?:
             throw ServiceConfigurationException(RequestDataFactory::class.java)
 
+        /**
+         * Creates a [RequestData] using the given strongly-typed parameters
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * Creates a [RequestData] using the given strongly-typed parameters.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 10]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         *
+         * @param requestId the client request which would be used by Corda to identify a particular request
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * @param requestId The client request which would be used by Corda to identify a particular request
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 11]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         * for a flow call
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * for a flow call.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 12]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         * @param flowClass the flow class to be constructed and called
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * @param flowClass The flow class to be constructed and called.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 13]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         * @param request data which will be serialized using a
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * @param request Data which will be serialized using a
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 14]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         * [net.corda.v5.application.marshalling.JsonMarshallingService] and passed to the flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * [net.corda.v5.application.marshalling.JsonMarshallingService] and passed to the flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 15]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         * @return a [RequestData] with properties that match the provided parameters
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * @return A [RequestData] with properties that match the provided parameters.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 16]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         */
         fun create(requestId: String, flowClass: Class<out Flow>, request: Any): RequestData
             = factory.create(requestId, flowClass, request)
 
+        /**
+         * Creates a [RequestData] using the given parameters. The strings used in this method are the same that
+         * would be passed to Swagger UI if using it.
+         *
+         * @param requestId the client request which would be used by Corda to identify a particular request
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * @param requestId The client request which would be used by Corda to identify a particular request
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 17]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         * for a flow call
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * for a flow call.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 18]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         * @param flowClass the name of the flow class to be constructed and called
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * @param flowClass The name of the flow class to be constructed and called.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 19]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         * @param request data to be passed to the flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * @param request Data to be passed to the flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 20]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         * @return a [RequestData] with the provided parameters as properties
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * @return A [RequestData] with the provided parameters as properties.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt [hunk 21]
```diff
@@ -6,24 +6,66 @@ import net.corda.v5.application.flows.Flow
...
+         */
         fun create(requestId: String, flowClass: String, request: String) : RequestData
             = factory.create(requestId, flowClass, request)
 
+        /**
+         * Creates a [RequestData] using the provided input. The input used in this method is the same that would
+         * be provided to `curl` if using it.
+         *
+         * @param jsonInput a Json-formatted string containing a client-provided `requestId`, the `flowClass` to
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/RequestData.kt]:
```suggestion
         * @param jsonInput A JSON-formatted string containing a client-provided `requestId`, the `flowClass` to
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt [hunk 22]
```diff
@@ -5,16 +5,35 @@ import net.corda.v5.application.flows.ResponderFlow
...
+ */
 @DoNotImplement
 interface SimulatedCordaNetwork : Closeable {
+
+    /**
+     * Checks the flow class and creates a simulated virtual node for the given holding identity with a list of
+     * [Flow] classes that the node can run. See [net.corda.simulator.tools.FlowChecker] for details of exceptions
+     * that can be thrown.
+     *
+     * @param holdingIdentity the holding identity for which to create the flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt]:
```suggestion
     * @param holdingIdentity The holding identity for which to create the flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt [hunk 23]
```diff
@@ -5,16 +5,35 @@ import net.corda.v5.application.flows.ResponderFlow
...
+     * @param flowClasses a list of flow classes to be checked
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt]:
```suggestion
     * @param flowClasses A list of flow classes to be checked.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt [hunk 24]
```diff
@@ -5,16 +5,35 @@ import net.corda.v5.application.flows.ResponderFlow
...
+     * @return a simulated virtual node in which flows can be run
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt]:
```suggestion
     * @return A simulated virtual node in which flows can be run.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt [hunk 25]
```diff
@@ -5,16 +5,35 @@ import net.corda.v5.application.flows.ResponderFlow
...
         holdingIdentity: HoldingIdentity,
         vararg flowClasses: Class<out Flow>
     ): SimulatedVirtualNode
 
     /**
-     * Creates a virtual node holding a concrete instance of a responder flow. Note that this bypasses all
+     * Creates a simulated virtual node holding a concrete instance of a responder flow. Note that this bypasses all
      * checks for constructor and annotations on the flow.
+     *
+     * @param responder the holding identity which will respond with this flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt]:
```suggestion
     * @param responder The holding identity which will respond with this flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt [hunk 26]
```diff
@@ -5,16 +5,35 @@ import net.corda.v5.application.flows.ResponderFlow
...
+     * @param protocol the protocol for which this responder instance should be run
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt]:
```suggestion
     * @param protocol The protocol for which this responder instance should be run.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt [hunk 27]
```diff
@@ -5,16 +5,35 @@ import net.corda.v5.application.flows.ResponderFlow
...
+     * @param responderFlow an instance of a responder flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt]:
```suggestion
     * @param responderFlow An instance of a responder flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt [hunk 28]
```diff
@@ -5,16 +5,35 @@ import net.corda.v5.application.flows.ResponderFlow
...
+     * @return a simulated virtual node which can run this instance of a responder flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedCordaNetwork.kt]:
```suggestion
     * @return A simulated virtual node which can run this instance of a responder flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedVirtualNode.kt [hunk 29]
```diff
@@ -20,12 +33,21 @@ interface SimulatedVirtualNode {
     fun callFlow(input: RequestData): String
 
     /**
-     * Retrieves the persistence service associated with this node's member
+     * @return the persistence service associated with this node
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedVirtualNode.kt]:
```suggestion
     * @return The persistence service associated with this node.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedVirtualNode.kt [hunk 30]
```diff
@@ -20,12 +33,21 @@ interface SimulatedVirtualNode {
...
 
     /**
-     * Generates a key for this node with the given alias, HSM category and scheme.
+     * Generates a key for this node which can then be accessed using the
+     * [net.corda.v5.application.membership.MemberLookup] service and used with the
+     * [net.corda.v5.application.crypto.SigningService] and
+     * [net.corda.v5.application.crypto.DigitalSignatureVerificationService]. Note that Simulator does not actually
+     * perform encryption, simulating it instead, and no private keys are held.
+     *
+     * @param alias an alias for the key
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedVirtualNode.kt]:
```suggestion
     * @param alias An alias for the key.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedVirtualNode.kt [hunk 31]
```diff
@@ -20,12 +33,21 @@ interface SimulatedVirtualNode {
...
+     * @param hsmCategory the HSM category for the key
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedVirtualNode.kt]:
```suggestion
     * @param hsmCategory The HSM category for the key.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatedVirtualNode.kt [hunk 32]
```diff
@@ -20,12 +33,21 @@ interface SimulatedVirtualNode {
...
+     * @param scheme the scheme for the key - not used, except in verification; an ECDSA key will be returned
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatedVirtualNode.kt]:
```suggestion
     * @param scheme The scheme for the key. This is only used in verification. An ECDSA key will be returned.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt [hunk 33]
```diff
@@ -33,12 +33,13 @@ class Simulator(
...
-     * Registers a flow class against a given member with Simulator. Flow classes will be checked
-     * for validity. Responder flows will also be registered against their protocols.
+     * Checks the flow class and creates a simulated virtual node for the given holding identity with a list of
+     * [Flow] classes that the node can run. See [net.corda.simulator.tools.FlowChecker] for details of exceptions
+     * that can be thrown.
      *
-     * @member The member for whom this node will be created
-     * @flowClasses The flows which will be available to run in the nodes. Must be `RPCStartableFlow`
-     * or `ResponderFlow`
+     * @param holdingIdentity the holding identity for which to create the flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt]:
```suggestion
     * @param holdingIdentity The holding identity for which to create the flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt [hunk 34]
```diff
@@ -33,12 +33,13 @@ class Simulator(
...
+     * @param flowClasses a list of flow classes to be checked
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt]:
```suggestion
     * @param flowClasses A list of flow classes to be checked.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt [hunk 35]
```diff
@@ -33,12 +33,13 @@ class Simulator(
...
+     * @return a simulated virtual node in which flows can be run
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt]:
```suggestion
     * @return A simulated virtual node in which flows can be run.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt [hunk 36]
```diff
@@ -48,12 +49,13 @@ class Simulator(
...
 
     /**
-     * Creates a virtual node holding a concrete instance of a responder flow. Note that this bypasses all
+     * Creates a simulated virtual node holding a concrete instance of a responder flow. Note that this bypasses all
      * checks for constructor and annotations on the flow.
      *
-     * @responder The member for whom this node will be created
-     * @protocol The protocol for which this flow will be called (does not need to be annotated on the flow)
-     * @responderFlow The instance of flow to be called in response to an initiating flow with the given protocol
+     * @param responder the holding identity which will respond with this flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt]:
```suggestion
     * @param responder The holding identity which will respond with this flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt [hunk 37]
```diff
@@ -48,12 +49,13 @@ class Simulator(
...
+     * @param protocol the protocol for which this responder instance should be run
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt]:
```suggestion
     * @param protocol The protocol for which this responder instance should be run.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt [hunk 38]
```diff
@@ -48,12 +49,13 @@ class Simulator(
...
+     * @param responderFlow an instance of a responder flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt]:
```suggestion
     * @param responderFlow An instance of a responder flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt [hunk 39]
```diff
@@ -48,12 +49,13 @@ class Simulator(
...
+     * @return a simulated virtual node which can run this instance of a responder flow
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/Simulator.kt]:
```suggestion
     * @return A simulated virtual node which can run this instance of a responder flow.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/SimulatorConfiguration.kt [hunk 40]
```diff
@@ -3,10 +3,25 @@ package net.corda.simulator
...
 import java.time.Duration
 
+/**
+ * Configuration for Simulator. This interface may be implemented directly or built using a
+ * [net.corda.simulator.factories.SimulatorConfigurationBuilder].
+ */
 interface SimulatorConfiguration {
 
+    /**
+     * The interval at which to check for responder flow health.
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/SimulatorConfiguration.kt]:
```suggestion
     * The interval at which to check responder flow health.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/HoldingIdentityFactory.kt [hunk 41]
```diff
@@ -1,8 +1,19 @@
...
+/**
+ * Creates the holding ID which is needed to create virtual nodes using Simulator. This interface should not
+ * be used directly; instead use the [net.corda.simulator.HoldingIdentity.create] methods on
+ * [net.corda.simulator.HoldingIdentity] itself.
+ */
+@DoNotImplement
 interface HoldingIdentityFactory {
+
+    /**
+     * @param memberX500Name the member for which to create a [HoldingIdentity]
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/HoldingIdentityFactory.kt]:
```suggestion
     * @param memberX500Name The member for which to create a [HoldingIdentity].
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/JsonMarshallingServiceFactory.kt [hunk 42]
```diff
@@ -1,16 +1,31 @@
...
+ */
+@DoNotImplement
 interface JsonMarshallingServiceFactory {
 
     companion object {
         private val delegate by lazy { ServiceLoader.load(JsonMarshallingServiceFactory::class.java).first().create() }
+
+        /**
+         * Creates a standalone service, which can be used outside of Simulator for
+         * parsing and formatting of arbitrary objects into Json.
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/JsonMarshallingServiceFactory.kt]:
```suggestion
         * parsing and formatting of arbitrary objects into JSON.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/JsonMarshallingServiceFactory.kt [hunk 43]
```diff
@@ -1,16 +1,31 @@
...
+         *
+         * @return a [JsonMarshallingService]
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/JsonMarshallingServiceFactory.kt]:
```suggestion
         * @return A [JsonMarshallingService].
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/JsonMarshallingServiceFactory.kt [hunk 44]
```diff
@@ -1,16 +1,31 @@
...
+         */
         fun create() : JsonMarshallingService {
             return delegate
         }
     }
 
+    /**
+     * @return a [JsonMarshallingService]
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/JsonMarshallingServiceFactory.kt]:
```suggestion
     * @return A [JsonMarshallingService].
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt [hunk 45]
```diff
@@ -1,25 +1,45 @@
...
- * Builds a configuration with which to call a flow, providing sensible defaults for any value which is
- * not explicitly overridden.
+ * Builds a [net.corda.simulator.SimulatorConfiguration] with which to call a flow, providing sensible defaults
+ * for any value which is not explicitly overridden.
  */
+@DoNotImplement
 interface SimulatorConfigurationBuilder {
 
+    /**
+     * @param clock the clock with which to build the configuration
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt]:
```suggestion
     * @param clock The clock with which to build the configuration.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt [hunk 46]
```diff
@@ -1,25 +1,45 @@
...
+     * @return a copy of this builder, with the given clock
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt]:
```suggestion
     * @return A copy of this builder, with the given clock.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt [hunk 47]
```diff
@@ -1,25 +1,45 @@
...
+     */
     fun withClock(clock: Clock): SimulatorConfigurationBuilder
 
+    /**
+     * @param pollInterval the interval at which to check for responder flow health
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt]:
```suggestion
     * @param pollInterval The interval at which to check for responder flow health.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt [hunk 48]
```diff
@@ -1,25 +1,45 @@
...
+     * @return a copy of this builder, with the given poll interval
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt]:
```suggestion
     * @return A copy of this builder, with the given poll interval.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt [hunk 49]
```diff
@@ -1,25 +1,45 @@
...
+     */
     fun withPollInterval(pollInterval: Duration): SimulatorConfigurationBuilder
 
+    /**
+     * @param timeout the maximum length of time to wait for a responder flow to respond
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt]:
```suggestion
     * @param timeout The maximum length of time to wait for a responder flow to respond.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt [hunk 50]
```diff
@@ -1,25 +1,45 @@
...
+     * @return a copy of this builder, with the given timeout
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt]:
```suggestion
     * @return A copy of this builder, with the given timeout.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt [hunk 51]
```diff
@@ -1,25 +1,45 @@
...
+     */
     fun withTimeout(timeout: Duration): SimulatorConfigurationBuilder
 
+    /**
+     * @return configuration built using chosen options or defaults if no overriding options have been selected
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt]:
```suggestion
     * @return Configuration built using chosen options or defaults if no overriding options have been selected.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt [hunk 52]
```diff
@@ -1,25 +1,45 @@
...
+     */
     fun build(): SimulatorConfiguration
 
     companion object {
+        /**
+         * @return a factory for building [net.corda.simulator.SimulatorConfiguration]
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorConfigurationBuilder.kt]:
```suggestion
         * @return A factory for building [net.corda.simulator.SimulatorConfiguration].
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorDelegateFactory.kt [hunk 53]
```diff
@@ -2,7 +2,20 @@ package net.corda.simulator.factories
...
+ * Simulator uses this factory to load a delegate implementation at runtime. This interface should not
+ * be used directly; construct a new [net.corda.simulator.Simulator] instead.
+ */
+@DoNotImplement
 interface SimulatorDelegateFactory {
+
+    /**
+     * Constructs a new Simulator delegate using the provided configuration or default if none provided.
+     *
+     * @param configuration the configuration with which to construct [net.corda.simulator.Simulator]
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorDelegateFactory.kt]:
```suggestion
     * @param configuration The configuration with which to construct [net.corda.simulator.Simulator].
```

File: simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorDelegateFactory.kt [hunk 54]
```diff
@@ -2,7 +2,20 @@ package net.corda.simulator.factories
...
+     * @return a [net.corda.simulator.Simulator] delegate
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/factories/SimulatorDelegateFactory.kt]:
```suggestion
     * @return A [net.corda.simulator.Simulator] delegate.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt [hunk 55]
```diff
@@ -2,14 +2,39 @@ package net.corda.simulator.tools
...
+/**
+ * Checks your [net.corda.v5.application.flows.Flow] for common mistakes with annotations and constructors.
+ */
+@DoNotImplement
 interface FlowChecker {
+
+    /**
+     * @param flowClass the [Flow] to check
+     * @throws [net.corda.simulator.exceptions.NoDefaultConstructorException] if no default constructor was found
+     * and was required
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt]:
```suggestion
     * and was required.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt [hunk 56]
```diff
@@ -2,14 +2,39 @@ package net.corda.simulator.tools
...
+     * @throws [net.corda.simulator.exceptions.NoProtocolAnnotationException] if no
+     * [net.corda.v5.application.flows.InitiatingFlow] or [net.corda.v5.application.flows.InitiatedBy] annotation was
+     * found with its accompanying protocol and was required
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt]:
```suggestion
     * found with its accompanying protocol and was required.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt [hunk 57]
```diff
@@ -2,14 +2,39 @@ package net.corda.simulator.tools
...
+     * @throws [net.corda.simulator.exceptions.UnrecognizedFlowClassException] if the flow does not inherit from a
+     * recognized flow class
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt]:
```suggestion
     * recognized flow class.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt [hunk 58]
```diff
@@ -2,14 +2,39 @@ package net.corda.simulator.tools
...
+     * @throws [net.corda.simulator.exceptions.NoSuspendableCallMethodException] if the flow is missing a
+     * [net.corda.v5.base.annotations.Suspendable] annotation on the `call` method
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt]:
```suggestion
     * [net.corda.v5.base.annotations.Suspendable] annotation on the `call` method.
```

File: simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt [hunk 59]
```diff
@@ -2,14 +2,39 @@ package net.corda.simulator.tools
...
 
     companion object {
         private val delegate = ServiceLoader.load(FlowChecker::class.java).first() ?:
             throw ServiceConfigurationException(FlowChecker::class.java)
+
+        /**
+         * Creates and runs a [FlowChecker] against the provided flow. See the [FlowChecker.check] method
+         * for details of possible exceptions thrown by this class.
+         *
+         * @param flowClass the [Flow] to check
```
Comment (nadinequinr3) [inline simulator/api/src/main/kotlin/net/corda/simulator/tools/FlowChecker.kt]:
```suggestion
         * @param flowClass The [Flow] to check.
```

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]

**Review by nadinequinr3 — APPROVED** [review]

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Pull request merged**